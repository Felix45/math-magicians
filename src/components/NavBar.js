import React from 'react';
import { NavLink } from 'react-router-dom';

const NavBar = () => {
  const urls = [
    { id: 1, text: 'Home', url: '/math-magician' },
    { id: 2, text: 'Calculator', url: '/calculator' },
    { id: 3, text: 'Quote', url: '/quote' },
  ];
  return (
    <nav className="d-flex space-between navbar">
      <header>
        <h1 className="title">Math Magician</h1>
      </header>
      <ul className="d-flex">
        {
          urls.map((url) => (
            <li
              key={url.id}
            >
              <NavLink to={url.url}>{ url.text }</NavLink>
            </li>
          ))
        }
      </ul>
    </nav>
  );
};

export default NavBar;
