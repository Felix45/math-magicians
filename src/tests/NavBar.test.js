import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import NavBar from '../components/NavBar';

it('renders correctly', () => {
  const tree = renderer
    .create(<BrowserRouter><NavBar /></BrowserRouter>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});