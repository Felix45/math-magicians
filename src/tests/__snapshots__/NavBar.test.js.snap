// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<nav
  className="d-flex space-between navbar"
>
  <header>
    <h1
      className="title"
    >
      Math Magicians
    </h1>
  </header>
  <ul
    className="d-flex"
  >
    <li>
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li>
      <a
        className=""
        href="/calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
    </li>
    <li>
      <a
        className=""
        href="/quote"
        onClick={[Function]}
      >
        Quote
      </a>
    </li>
  </ul>
</nav>
`;
